### PR TITLE
Added a check for whether or not a git repository exists, and option …

### PR DESCRIPTION
**Pull request by vondenstein**
## Description

Added a check for whether or not a git repository exists, and option to continue if this check fails.

### Documentation
Updated README to include documentation for continueWithoutRepo flag.

## Related Issues

Addresses #9

### CODE REVIEW

File: src/gatsby-node.js [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+  activity.start()
+
+  const exists = await simpleGit().checkIsRepo()
+
+  if (!exists) {
+    ;(pluginOptions.continueWithoutRepo
+      ? reporter.error
+      : reporter.panicOnBuild)(
+      "No git repository found.",
+      Error("fatal: not a git repository (or any parent up to mount point /)")
```
Comment (Mattadore) [inline src/gatsby-node.js:19]: nice ternary bro

File: src/gatsby-node.js [hunk 2]
```diff
@@ -1,9 +1,34 @@
 const simpleGit = require(`simple-git`)
 import { pluginOptionsSchema } from "./options-validation"
+import { getDate } from "./repo"
```
Comment (Mattadore) [inline src/gatsby-node.js]: Where is ./repo tho

File: src/gatsby-node.js [hunk 3]
```diff
@@ -14,15 +39,23 @@ exports.onCreatePage = async ({ page, actions }, pluginOptions) => {
...
+  const repoExists = await cache.get(cacheKey)
+
+  let lastMod
+  if (repoExists) {
+    lastMod = await simpleGit().log({
+      file: filePath,
+      maxCount: 1,
+      strictDate: true,
+    })
+    lastMod = await getDate(filePath, reporter)
```
Comment (Mattadore) [inline src/gatsby-node.js]: 🤨

File: src/gatsby-node.js (hunk 2 above)
Comment (vondenstein) [reply src/gatsby-node.js]: Whoops, left that in from testing. Nice catch.

**Review by Mattadore — APPROVED** [review]
LGTM

**Pull request merged**